### PR TITLE
Unsteady debug statements

### PR DESCRIPTION
**Pull request by bburke38**
- Create debug flags for fun3d_interface and nlbgs driver
- Print out some adjoint variables, struct_loads, aero_loads, etc.

### CODE REVIEW

**Comment by sean-engelstad** [comment]
Can you make it so we only print out the real and imaginary norms, not the vector? I don't think the vector is helpful. Maybe something like this. And maybe we can save the definition of real and imaginary norms in one of the utils python modules or something so you can use it in different files
```
def real_norm(vec):
    return np.max(np.abs(np.real(vec))) # or do a np.linalg.norm either one
def imag_norm(vec):
    return np.max(np.abs(np.imag(vec)))
if struct_loads is not None:
    print(f"struct loads real step{step} = {real_norm(struct_loads)}")
    print(f"struct loads imag step{step} = {imag_norm(struct_loads)}", flush=True)
```

**Comment by sean-engelstad** [comment]
Also why is the test failing now?

**Comment by bburke38** [comment]
I missed some checks for none types, I'll add those this afternoon.

**Comment by bburke38** [comment]
I added a None check inside the new utilities functions. That way, if the array is None it automatically will print out NoneType instead of throwing an error from linalg.norm.

File: funtofem/driver/funtofem_nlbgs_driver.py [hunk 1]
```diff
@@ -157,6 +160,22 @@ def _solve_steady_forward(self, scenario, steps=None):
                 body.transfer_loads(scenario)
                 body.transfer_heat_flux(scenario)
 
+                if self._debug:
```
Comment (sean-engelstad) [inline funtofem/driver/funtofem_nlbgs_driver.py:163]: You need a self.comm.rank == 0 check here so for large proc cases you don't print this out 48 times, etc. If you do want it to print out for every processor then we should have a different debug mode like `debug = "all"`, `debug = "root"` and `debug=None` or something.
```suggestion
                if self._debug and self.comm.rank == 0:
```

File: funtofem/driver/funtofem_nlbgs_driver.py [hunk 2]
```diff
@@ -289,6 +308,23 @@ def _solve_unsteady_forward(self, scenario, steps=None):
                 body.transfer_loads(scenario, time_index)
                 body.transfer_heat_flux(scenario, time_index)
 
+                if self._debug:
```
Comment (sean-engelstad) [inline funtofem/driver/funtofem_nlbgs_driver.py:311]: See above, comm check needed here, I'm not going to repeat this comment for the other ones.

**Review by sean-engelstad — CHANGES_REQUESTED** [review]
I think we should add some root proc checks in case we are debugging in the large proc case. As far as I can tell, I think printing out on the root proc only is good (only in rare cases would we need to print out the norms for each individual proc).

**Review by sean-engelstad — APPROVED** [review]
Comm checks were added. Approved

**Pull request merged**